client/components/data/query-site-monitor-settings/index.jsx

```diff
@@ -0,0 +1,50 @@
+/**
+ * External dependencies
+ */
+import { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import { isRequestingSiteMonitorSettings } from 'state/selectors';
+import { requestSiteMonitorSettings } from 'state/sites/monitor/actions';
+
+class QuerySiteMonitorSettings extends Component {
+	static propTypes = {
+		siteId: PropTypes.number.isRequired,
+		requestingSiteMonitorSettings: PropTypes.bool,
+		requestSiteMonitorSettings: PropTypes.func
+	};
+
+	componentWillMount() {
+		this.request( this.props );
+	}
+
+	componentWillReceiveProps( nextProps ) {
+		if ( this.props.siteId !== nextProps.siteId ) {
+			this.request( nextProps );
+		}
+	}
+
+	request( props ) {
+		if ( props.requestingSiteMonitorSettings ) {
```

Comment:
> Let's also check if siteId is falsey for the early return

Good call, done in https://github.com/Automattic/wp-calypso/pull/13169/commits/d06adc9f26ce7b71bfedca2f90ed16873b048339.

> We can maybe save a network request if we check if the site is a jetpack site, but optional since the request just returns an error in that case.

I suggest that we skip that. Who would want to use that query component for a non-jetpack site anyway? I could add the check if you feel it will be necessary though.